packages/grafana-ui/src/components/ConfirmModal/ConfirmModal.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { FC, useContext } from 'react';
+import { css } from 'emotion';
+import { Modal } from '../Modal/Modal';
+import { IconType } from '../Icon/types';
+import { Button } from '../Button/Button';
+import { stylesFactory, ThemeContext } from '../../themes';
+import { GrafanaTheme } from '@grafana/data';
+
+const getStyles = stylesFactory((theme: GrafanaTheme) => ({
+  modal: css`
+    width: 500px;
+  `,
+  modalContent: css`
+    text-align: center;
+  `,
+  modalText: css`
+    font-size: ${theme.typography.heading.h4};
+    color: ${theme.colors.link};
+    margin-bottom: calc(${theme.spacing.d} * 2);
+    padding-top: ${theme.spacing.d};
+  `,
+  modalButtonRow: css`
+    margin-bottom: 14px;
+    a,
+    button {
+      margin-right: ${theme.spacing.d};
+    }
+  `,
+}));
+
+const defaultIcon: IconType = 'exclamation-triangle';
+
+interface Props {
+  isOpen: boolean;
+  title: string;
+  body: string;
+  confirmText: string;
+  icon?: IconType;
+
+  onConfirm(): void;
+  onDismiss(): void;
+}
+
+export const ConfirmModal: FC<Props> = ({ isOpen, title, body, confirmText, icon, onConfirm, onDismiss }) => {
+  const theme = useContext(ThemeContext);
+  const styles = getStyles(theme);
+
+  return (
+    <Modal className={styles.modal} title={title} icon={icon || defaultIcon} isOpen={isOpen} onDismiss={onDismiss}>
```

Comment:
Nit: instead of passing `isOpen` property, could we assume isOpen = true here and only render the ConfirmModal when necessary?

```tsx
shouldConfirm && <ConfirmModal {...} />
```

Reply:
Something like:
```ts
export const ConfirmModal: FC<Props> = ({ isOpen, title, body, confirmText, icon, onConfirm, onDismiss }) => {
  const theme = useContext(ThemeContext);
  const styles = getStyles(theme);

  return (
    isOpen && <Modal className={styles.modal} title={title} icon={icon || defaultIcon} isOpen={true} onDismiss={onDismiss}>
  ...
```
?

Reply:
I was thinking more abutt controlling the modal being rendered or not externally.

So, wherever you would like to render the modal, I assume it's a ConfirmButton or sth like this, you would do:

```tsx
shouldConfirm && <ConfirmModal {...} />
```

instead of:

```tsx
<ConfirmModal isOpen={isOpen} {...} />
```

Reply:
Ok, I see now. From my point of view it looks strange a little bit because base `Modal` component has that property.

Reply:
Yeah, that's right. I'm usually opt for not rendering components rather than returning nulls from render. So this should be fixed in the Modal, rather than here. I'll leave the final decision with you.

Reply:
Modal does not render anything when isOpen = false, think this can follow same pattern